docs/source/migrations/modes.rst

```diff
@@ -0,0 +1,50 @@
+======================
+Snuba Migration Modes
+======================
+
+This doc outlines a basic way to try out distributed and local migrations.
+Note that this is experimental, and should be used only for development
+purposes at the moment.
+
+The main "switch" between the two modes for running data migrations (local and
+distributed) lives in ``sentry/conf/server.py``.
+The controlling envrionment variable is ``SENTRY_DISTRIBUTED_CLICKHOUSE_TABLES``,
+and its value must be set in order to use a specific mode.
+
+Once this boolean variable is set, one of two Clickhouse Docker volumes will be
+used for data storage, depending on the mode (distributed or local). Whenever a user
+wants to switch between the two modes, they must "turn off" the running Clickhouse
+container, alter the environment variable mentioned above, and then "turn on" the
+same container to be in the new mode.
+
+More information on migrations in general can be found `here <https://github.com/getsentry/snuba/blob/master/MIGRATIONS.md>`_.
+
+Enabling Local Mode
+=====================
+
+In your local ``server.py``, set ``SENTRY_DISTRIBUTED_CLICKHOUSE_TABLES``
+to False. Start up the corresponding Clickhouse container (``sentry devservices up clickhouse``).
+
+Ensuring that your local configuration is prepared for migrations (reverting any existing
+migrations, or dropping certain databases/tables), run migrations as expected
+(``snuba migrations migrate --force``).
+
+
+Enabling Distributed Mode
+============================
+
+In your local ``server.py``, set ``SENTRY_DISTRIBUTED_CLICKHOUSE_TABLES``
+to True. Start up the corresponding Clickhouse container (``sentry devservices up clickhouse``).
+
+Now, it is important to configure the clusters that can be used in Distributed tables. These are
+set in `this file <https://github.com/getsentry/sentry/blob/master/config/clickhouse/dist_config.xml>`_.
```

Comment:
The way this is worded suggests that I need to change something in this file before distributed mode will work. Is this true? Or is the default configuration already good enough?

Reply:
Thank you for raising this point - default configuration already works, and is probably best to work with at the moment. It is also possible, though, to write dist_config.xml such that there is more than 1 shard and replica, though that would also require understanding which storages can be sharded and which cannot be (will put out a PR to start some work on this soon). I'll edit the doc so these are noted.